### ISSUE TITLE
Isn't cwd a filesystem property, not a path propery?

### ISSUE THREAD

**Issue by altendky** [Julian/Filesystems#24]
It seems to me that the current working directory would be a property of a `filesystem`, not a `Path`.  Each filesystem could have it's own working directory.  On Windows, each drive actually has its own working directory...  but anyways, whatever is up with that needs to be covered with finishing out Windows support.

**Comment by Julian** [comment]
Yeah, this is probably a thing that'd need to change as we bring on windows support -- in windows it really is per-filesystem, or is it per-filesystem-per-environment?

On the *nix side, the idea here was "gimme a cwd for where I am right now in my current session" -- i.e. "what does . refer to right now"?

In theory that applies to Windows too though right, because the user is sitting on one of the filesystems right this second (i.e. what does `os.path.expand(os.curdir)` do on Windows?) but there's *also* the other idea of some "remembered" cwd on each filesystem right?

**Comment by altendky** [comment]
Windows has it per drive, yes.  Also an overall perhaps, I'll have to dig.  Though I meant the filesystem class.  It came to mind as I was going to copy something with a memory filesystem and I realized I would be getting the native working directory.
